### PR TITLE
fetch pending transactions for geth

### PR DESCRIPTION
**Pull request by ayrat555**
fixes https://github.com/poanetwork/blockscout/issues/2239

## Changelog

- add pending transactions for geth

### CODE REVIEW

File: apps/ethereum_jsonrpc/lib/ethereum_jsonrpc/geth.ex [hunk 1]
```diff
@@ -46,12 +46,31 @@ defmodule EthereumJSONRPC.Geth do
...
   @doc """
-  Pending transaction fetching is not supported currently for Geth.
-
-  To signal to the caller that fetching is not supported, `:ignore` is returned.
+  Fetches the pending transactions from the Geth node.
   """
   @impl EthereumJSONRPC.Variant
-  def fetch_pending_transactions(_json_rpc_named_arguments), do: :ignore
+  def fetch_pending_transactions(json_rpc_named_arguments) do
+    with {:ok, transaction_data} <-
```
Comment (pasqu4le) [inline apps/ethereum_jsonrpc/lib/ethereum_jsonrpc/geth.ex]: There is no `else` for this `with` block

Comment (ayrat555) [reply apps/ethereum_jsonrpc/lib/ethereum_jsonrpc/geth.ex]: the error will be returned to client

**Comment by ayrat555** [comment]
@vbaranov which chain did you use?

**Comment by vbaranov** [comment]
> @vbaranov which chain did you use?

Rinkeby, Geth 1.9.0

**Comment by ayrat555** [comment]
@vbaranov pending transactions weren't being inserted because of db constraint. (index have to be nil if block hash is nil). I fixed it

File: apps/ethereum_jsonrpc/lib/ethereum_jsonrpc/geth.ex [hunk 2]
```diff
@@ -63,9 +63,9 @@ defmodule EthereumJSONRPC.Geth do
         |> Transactions.to_elixir()
         |> Transactions.elixir_to_params()
         |> Enum.map(fn params ->
-          # txpool_content always returns transaction with 0x0000000000000000000000000000000000000000000000000000000000000000 value in block hash.
+          # txpool_content always returns transaction with 0x0000000000000000000000000000000000000000000000000000000000000000 value in block hash and index is not null.
```
Comment (vbaranov) [inline apps/ethereum_jsonrpc/lib/ethereum_jsonrpc/geth.ex]: @ayrat555 shouldn't it be changed to `and index is null`?

Comment (ayrat555) [reply apps/ethereum_jsonrpc/lib/ethereum_jsonrpc/geth.ex]: you're right. fixed

**Pull request merged**